### PR TITLE
Allow to override concrete --postgres-version/--mysql-version

### PR DESCRIPTION
**Pull request by potiuk**
The `BACKEND_VERSION` env variable is now generic way to override any version of any backend. When you set it before running breeze and when the value maches possible options for given platform, the value of concrete backend version will be overridden from the variable.

This will be helpful to generalize a bit and make reusable test workflow where `BACKEND_VERSION` will be treated in simiar way in all tests.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by Taragolis** [comment]
Hmm, I think now MySQL/Postgres tests is gone 🤔

**Comment by potiuk** [comment]
Yep. One more thing to add later today :). But I am close to get rid of 1300 or so lines of CI.yml after that one :)

File: dev/breeze/src/airflow_breeze/utils/selective_checks.py [hunk 1]
```diff
@@ -516,7 +516,7 @@ def postgres_exclude(self) -> list[dict[str, str]]:
             return []
         return [
             # Exclude all combinations that are repeating python/postgres versions
-            {"python-version": python_version, "postgres-version": postgres_version}
+            {"python-version": python_version, "backend-version": postgres_version}
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/utils/selective_checks.py:519]: That was the reason why we did not have postgres/mysql tests  in the first version.

**Comment by potiuk** [comment]
I also added a bit of documentation explaining the ``BACKEND_VERSION`` override behaviour as it was a little `magical`

**Comment by vincbeck** [comment]
> I also added a bit of documentation explaining the `BACKEND_VERSION` override behaviour as it was a little `magical`

Thanks for doing that, I did not quite understand the PR before reading it

**Review by vincbeck — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Review by dirrao — APPROVED** [review]

**Pull request merged**